templates/accounts.go

```diff
@@ -58,26 +57,42 @@ func exportType(t sema.Type) cadence.Type {
 }
 
 func newSignAlgoValue(sigAlgo crypto.SignatureAlgorithm) cadence.Enum {
-	sigAlgoCadence := sema.SignatureAlgorithmECDSA_P256
-	if sigAlgo == crypto.ECDSA_secp256k1 {
-		sigAlgoCadence = sema.SignatureAlgorithmECDSA_secp256k1
+	sigAlgoValue := sema.SignatureAlgorithmECDSA_P256
+	switch sigAlgo {
+	case crypto.ECDSA_P256:
+		sigAlgoValue = sema.SignatureAlgorithmECDSA_P256
+	case crypto.ECDSA_secp256k1:
+		sigAlgoValue = sema.SignatureAlgorithmECDSA_secp256k1
+	default:
+		panic(fmt.Sprintf("unsupported signature algorithm: %v", sigAlgo))
```

Comment:
should we return an error instead?

Reply:
we would have to propagate that error out of `AccountKeyToCadenceCryptoKey` then. Hm... yes, I guess It would be more proper. Let me see how big of a change this is.

Reply:
Yeah, is that used anyplace else? If it's too much of changes then maybe it can stay.

Reply:
I actually added it it in the emulator, e2e tests, flow-go and flow-core-contracts :). However for correctness sake I think it's better to change it.

Reply:
what do you think d263b74423232e9e10d88059dc86f4bfac8b02a0